src/lifetime-elision.md

```diff
@@ -129,6 +137,27 @@ struct TwoBounds<'a, 'b, T: ?Sized + 'a + 'b>
 TwoBounds<'a, 'b, dyn Foo<'c>>
 ```
 
+Here is an example featuring associated type bindings:
+
+```rust,ignore
+trait Bar<'a> {
+  type Item1: ?Sized;
+  type Item2: ?Sized + 'a;
+}
+trait Baz { }
+
+dyn Bar<
+  'x
+  Item1 = dyn Baz, // defaults to `dyn Baz + 'static`
+  Item2 = dyn Baz, // defaults to 'dyn Baz + `x`
```

Comment:
This is a great *and more complicated* (which is why it is great) example to include as a test in the PR!